src/python_routing_v02/fast_reach/mc_reach.pyx

```diff
@@ -375,11 +390,16 @@ cpdef object compute_network(
                 if assume_short_ts:
                     quc = qup
 
-                compute_reach_kernel(qup, quc, reachlen, buf_view, out_view, assume_short_ts)
+                compute_reach_kernel(qup, quc, reachlen, buf_view, out_view, assume_short_ts, return_courant)
 
                 # copy out_buf results back to flowdepthvel
                 for i in range(3):
                     fill_buffer_column(drows, i, srows, ts_offset + i, out_view, flowveldepth)
+                    
+                # copy out_buf results back to courant
+                if return_courant:
+                    for i in range(3,6):
+                        fill_buffer_column(drows, i, srows, ts_offset + (i-3), out_view, courant)
```

Comment:
If Courant metrics are requested, `fill_buffer_column` is used to write data from `out_view` to `courant`